File: quack/Quacks.java
package quack;

public class Quacks {
    private double QPS; //quacks per second
    private double QPC; //quacks per click
    private double multiplier;
    private int quacks;

    public Quacks() {
        QPS = 0;
        QPC = 1;
        multiplier = 1;
        quacks = 0;
    }

    /**
     * constrcutor for quack
     * @param QPS quacks per second
     * @param QPC quacks per click
     * @param multiplier the QPS = multiplier * QPS // update later once we actually figure it out
     */
    public Quacks(double QPS, double QPC, double multiplier) {
        this.QPS = QPS;
        this.QPC = QPC;
        this.multiplier = multiplier;
    }

    //getters and setters
    public double getQPS() {
        return QPS;
    }
    public void incrementQPS() {
        QPS;
    }
    public double getQPC() {
        return QPC;
    }
    public void setQPC(double qpc) {
        QPC = qpc;
    }
    public double getMultiplier() {
        return multiplier;
    }
    // public void multiplier(double multiplier) {
    //     this.multiplier = multiplier;
    // }
    public int getQuacks() {
        return quacks;
    }
    public void addQuacks(int quacks) {
        this.quacks += quacks;
    }
    public void removeQuacks(int quacks) {
        this.quacks -= quacks;
    }
    public void incrementQuacks() {
        quacks += QPC;
    }
}

File: qcActual/Quacks.js
class Quacks {
  // Constructor without parameters
  constructor(QPS = 0, QPC = 1, multiplier = 1) {
    this.QPS = QPS; // quacks per second
    this.QPC = QPC; // quacks per click
    this.multiplier = multiplier; 
    this.quacks = 0;
  }

  // click
  click() {
    this.quacks += this.QPC;
  }

  // Getters and setters
  getQPS() {      
    return this.QPS;
  }
  setQPS(qps) {
    this.QPS = qps;
  }
  incrementQPS() {
    this.QPS++;
    const doc = document.getElementById("amount_of_quacks_per_second");
    doc.innerHTML = "<p>Quacks Per Second: " + this.QPS + "</p>";
  }
  getQPC() {
    return this.QPC;
  }
  setQPC(qpc) {
    this.QPC = qpc;
  }
  getMultiplier() {
    return this.multiplier;
  }
  setMultiplier(multiplier) {
    this.multiplier = multiplier;
  }
  getQuacks() {
    return this.quacks;
  }
  addQuacks(quacks) {
    this.quacks += quacks;
  }
  removeQuacks(quacks) {
    this.quacks -= quacks;
  }
  updateDisplay() {    
    const doc = document.getElementById("amount_of_quacks");
    doc.innerHTML = "<h2>Number Quacks: " + Math.floor(this.quacks) + "</h2>";
  }
}